### PR TITLE
[fix][ci][branch-2.11] Fix shell license issue

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

Fix pulsar shell license issue.

### Modification
- Move `distribution/server/licenses` to `distribution/licenses`.
- Modify `server` and `shell` license to reference the parent path.
- Modify `check-binary-license.sh` filter `pulsar-client`/`pulsar-common`/`pulsar-websocket` jars.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: .github/workflows/pulsar-ci.yaml [hunk 1]
```diff
@@ -620,10 +620,14 @@ jobs:
...
           # build docker image
           # include building of Pulsar SQL, Connectors, Offloaders and server distros
-          mvn -B -am -pl pulsar-sql/presto-distribution,distribution/io,distribution/offloaders,distribution/server,tests/docker-images/latest-version-image install \
+          mvn -B -am -pl pulsar-sql/presto-distribution,distribution/io,distribution/offloaders,distribution/server,distribution/shell,tests/docker-images/latest-version-image install \
           -Pmain,docker -Dmaven.test.skip=true -Ddockerfile.build.squash=true \
           -Dspotbugs.skip=true -Dlicense.skip=true -Dcheckstyle.skip=true -Drat.skip=true
 
+      # check full build artifacts licenses
+      - name: Check binary licenses
+        run: src/check-binary-license.sh ./distribution/server/target/apache-pulsar-*-bin.tar.gz && src/check-binary-license.sh ./distribution/shell/target/apache-pulsar-shell-*-bin.tar.gz
```
Comment (mattisonchao) [inline .github/workflows/pulsar-ci.yaml:629]: `apache-pulsar-*-bin.tar.gz ` why * can't match `apache-pulsar-shell-*-bin.tar.gz`

Comment (Technoboy-) [reply .github/workflows/pulsar-ci.yaml:629]: Oh, I think it can match, but only in a different path.

Comment (mattisonchao) [reply .github/workflows/pulsar-ci.yaml:629]: Ah, sorry I ignored it. can we give the path a `*`?

**Review by mattisonchao — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]

File: .github/workflows/pulsar-ci.yaml (hunk 1 above)
Comment (Technoboy-) [reply .github/workflows/pulsar-ci.yaml:629]: Put a `*` on the path will match offloader and io, offloader and io are `nar`, license check with `nar` has to do more different things, we can separate this part.

**Pull request merged**